python/src/deltachat/provider.py

```diff
@@ -0,0 +1,37 @@
+""" Provider info class """
+
+from .capi import ffi, lib
+from .cutil import as_dc_charpointer, from_dc_charpointer
+
+
+class Provider(object):
+    def __init__(self, provider_instance):
+        self._provider = provider_instance
+
+    @classmethod
+    def from_email(cls, email):
+        instance = lib.dc_get_provider_info_from_email(as_dc_charpointer(email))
+        if instance == ffi.NULL:
+            return None
+        else:
+            return cls(instance)
+
+    @property
+    def overview_page(self):
+        return from_dc_charpointer(lib.dc_provider_get_overview_page(self._provider))
```

Comment:
These all should do the ffi.NULL check afaik.

Reply:
What does this mean? I have no experience with python/C/Rust integration.

Reply:
`lib.dc_provider_get_overview_page()` can return `NULL` (aka `ptr::null_mut()`) as an error code.  The python bindings should not just pass the error return to `from_dc_charpointer()` since that would just propagate it to the caller and that's very un-pythonic, instead it needs to check if the return value is `ffi.NULL` and if so needs to raise an appropriate exception.

Reply:
Currently, it returns only "" if provider was null but it returns never null itself (although you proposed to change this).

Anyway, as _provider is never null, dc_provider_get_overview_page never returns an error code. It is only an accessor function.

I'll add info about nullability to the docs later.

Reply:
Is it enough if I add this once at the beginning or shall I add this hint to every function? (Strings must be freed, dc_provider not)